src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs

```diff
@@ -71,7 +72,7 @@ private async Task<IOperationResult> DoBulkInsert(Action<IOperationProgress> onP
                                     // if we are going to wait on the network, flush immediately
                                     if ((task.Wait(5) == false && numberOfCommands > 0) ||
                                         // but don't batch too much anyway
-                                        totalSize > 16 * Voron.Global.Constants.Size.Megabyte)
+                                        totalSize > 16 * Voron.Global.Constants.Size.Megabyte || operationsCount >= 8192)
```

Comment:
Do we have some numbers for this? What if this is 4/8/16/32/64/128k ?

Reply:
yes I tested bulk insert of counter, ts, docs, for various values (1k to 20k), 8k gives the fastest result